pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/ui/player/mediacontroller/MediaControllerViewModel.kt

```diff
@@ -0,0 +1,172 @@
+        val listItems = mutableListOf<MediaItem>()
+        for (playlist in playlists) {
+            // In our case, children are the playlists so children are the playable content
+            listItems += getChildren(mediaBrowser, playlist.mediaId)
+        }
+        return listItems
+    }
+
+    private suspend fun getChildren(mediaBrowser: MediaBrowser, parentId: String): ImmutableList<MediaItem> {
+        return mediaBrowser.getChildren(parentId, 0, PAGE_SIZE, null).await().value ?: ImmutableList.of()
+    }
+
+    /**
+     * Play item
+     *
+     * @param mediaItem
+     */
+    fun playItem(mediaItem: MediaItem) {
+        var isInPlaylist = false
+        for (i in 0 until getPlayer().mediaItemCount) {
+            if (getPlayer().getMediaItemAt(i).mediaId == mediaItem.mediaId) {
+                getPlayer().seekToDefaultPosition(i)
+                isInPlaylist = true
+                break
+            }
+        }
+        if (!isInPlaylist) {
+            addItemToPlaylist(mediaItem, true)
+        }
+    }
+
+    /**
+     * Add item to playlist
+     *
+     * @param mediaItem MediaIterm to add
+     * @param autoPlay AutoPlay set to true to start playback too
+     */
+    fun addItemToPlaylist(mediaItem: MediaItem, autoPlay: Boolean = false) {
+        getPlayer().addMediaItem(mediaItem)
+        if (getPlayer().playbackState == Player.STATE_IDLE || getPlayer().playbackState == Player.STATE_ENDED) {
+            getPlayer().prepare()
+        }
+        if (autoPlay) {
+            getPlayer().seekToDefaultPosition(getPlayer().mediaItemCount - 1)
+            getPlayer().play()
+        }
+    }
+
+    /**
+     * Remove item from the playlist
+     *
+     * @param mediaItem MediaItem to remove
+     */
+    fun removeItem(mediaItem: MediaItem) {
+        val itemCount = getPlayer().mediaItemCount
+        for (i in 0 until itemCount) {
+            if (getPlayer().getMediaItemAt(i).mediaId == mediaItem.mediaId) {
+                getPlayer().removeMediaItem(i)
+                break
+            }
+        }
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        player.value?.removeListener(listener)
+        controllerConnection.release()
+    }
+
+    private fun getPlayer(): Player {
+        return player.value!!
+    }
+
+    private fun getPlayerListItems(player: Player): List<MediaItem> {
+        val mediaList = mutableListOf<MediaItem>()
+        val itemCount = player.mediaItemCount
+        for (i in 0 until itemCount) {
+            mediaList += player.getMediaItemAt(i)
+        }
+        return mediaList
+    }
+
+    private inner class ComponentListener : Player.Listener {
+        override fun onMediaItemTransition(mediaItem: MediaItem?, reason: Int) {
+            _currentPlayingItem.value = mediaItem ?: MediaItem.EMPTY
+        }
+
+        override fun onTimelineChanged(timeline: Timeline, reason: Int) {
+            _currentPlaylistItems.value = getPlayerListItems(getPlayer())
```

Comment:
⚠️ **[Android Lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SyntheticAccessor: Access to `private` method `getPlayer` of class `MediaControllerViewModel` requires synthetic accessor